src/Utilities.cmake

```
@@ -124,3 +124,48 @@ function(detect_architecture arch)
     set(${arch} x64 PARENT_SCOPE)
   endif()
 endfunction()
+
+# detect sanitizers support for compiler
+function(check_sanitizers_support
+        ENABLE_SANITIZER_ADDRESS
+        ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+        ENABLE_SANITIZER_LEAK
+        ENABLE_SANITIZER_THREAD
+        ENABLE_SANITIZER_MEMORY)
+  set(SANITIZERS "")
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
```

The tests are failing because the sanitizers are not supported on Windows with Clang and MinGW (out of the box).

```suggestion
if(NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
```